src/Versioning/ChangesetSnapshotModel.cs

```diff
@@ -79,6 +79,48 @@ public enum ChangesetSnapshotActionType
         Review
     }
 
+    /// <summary>
+    /// Contains an enumerated list of available changeset snapshot process types.
+    /// </summary>
+    [JsonConverter(typeof(StringEnumConverter))]
+    public enum ChangesetSnapshotProcessType
+    {
+        /// <summary>
+        /// A component was created new from template.
+        /// </summary>
+        CreateNew,
```

Comment:
JsonConverter would serialize these values to strings, but to pass these values to those functions I'm not sure if we'd still need to use the EnumMember() decorator, otherwise they fall back to integer values starting from 0.

[EnumMember(Value = "CreateNew")]
CreateNew

Reply:
I don't think its necessary. I use .ToString() in the backend yet leave the methods as string arguments.  That way, we can make it easy on ourselves yet allow the user to enter any custom string they want.

Reply:
Sure, please make sure the string values get resolved from those enums. By default, they yield numeric values.